src/GafferImage/Display.cpp

```diff
@@ -268,42 +299,54 @@ class GafferDisplayDriver : public IECoreImage::DisplayDriver
 			Display::driverCreatedSignal()( driver.get(), parameters.get() );
 		}
 
-		ConstFloatVectorDataPtr getTile( const V2i &tileOrigin, size_t channelIndex )
+		struct TileCacheElement
 		{
-			V2i tileIndex = tileOrigin / ImagePlug::tileSize();
-
-			if(
-				tileIndex.x < m_tiles.index_bases()[0] ||
-				tileIndex.x >= (int)(m_tiles.index_bases()[0] + m_tiles.shape()[0] ) ||
-				tileIndex.y < m_tiles.index_bases()[1] ||
-				tileIndex.y >= (int)(m_tiles.index_bases()[1] + m_tiles.shape()[1] )
-			)
+			TileCacheElement(): backBufferData( ImagePlug::blackTile()->copy() ), backBuffer( backBufferData->writable() ), dirty( false ), cachedTile( ImagePlug::blackTile() ), cachedForDataCount( 0 )
 			{
-				// outside data window
-				return nullptr;
 			}
 
-			tbb::spin_rw_mutex::scoped_lock tileLock( m_tileMutex, false /* read */ );
+			TileCacheElement( const TileCacheElement& other ) : TileCacheElement()
+			{
+				*this = other;
+			}
 
-			ConstFloatVectorDataPtr result = m_tiles[tileIndex.x][tileIndex.y][channelIndex];
-			if( !result )
+			TileCacheElement& operator=( const TileCacheElement& other )
 			{
-				result = ImagePlug::blackTile();
+				memcpy( &backBuffer[0], &other.backBuffer[0], backBuffer.size() * sizeof( float ) );
+				dirty = (bool)other.dirty;
+				cachedTile = other.cachedTile;
+				cachedForDataCount = other.cachedForDataCount;
```

Comment:
Shouldn't we be holding a lock when accessing these? Oh, is that being held for us in the GafferDisplayDriver copy constructor? That makes me realise that there's a single lock for all tile accesses - can we reduce the granularity and lock per tile instead?